### PR TITLE
Rename delete_all to schedule_detele_all

### PR DESCRIPTION
**Pull request by sadhansood**
## Description 

Rename delete_all in typed store library to schedule_delete_all and add more documentation around it.

## Test Plan 
Existing tests

### CODE REVIEW

**Review by tzakian — APPROVED** [review]
This looks good to me, thank you for the name change and comment!

**Comment by mystenmark** [comment]
I think we might want similar comments (and possibly a similar rename) for `delete_range`

**Comment by sadhansood** [comment]
> I think we might want similar comments (and possibly a similar rename) for `delete_range`

good point, let me add that in this PR itself

File: crates/sui-core/src/authority/authority_store_tables.rs [hunk 1]
```diff
@@ -392,7 +392,7 @@ impl AuthorityPerpetualTables {
         wb: &mut DBBatch,
         object: &ObjectKey,
     ) -> SuiResult<ObjectRef> {
-        wb.delete_range(
+        wb.schedule_delete_range(
```
Comment (lxfind) [inline crates/sui-core/src/authority/authority_store_tables.rs:395]: So, is this actually safe?

Comment (sadhansood) [reply crates/sui-core/src/authority/authority_store_tables.rs:395]: It should be, this table is configured to not return old values post delete

**Pull request merged**